### PR TITLE
Port20048 2.5

### PR DESCRIPTION
**Pull request by RyanCavanaugh**
Ports #20048 (and its dependents) to release-2.5

### CODE REVIEW

File: tests/baselines/reference/api/tsserverlibrary.d.ts [hunk 1]
```diff
@@ -0,0 +1,7692 @@
+/*! *****************************************************************************
```
Comment (billti) [inline tests/baselines/reference/api/tsserverlibrary.d.ts]: Why is this file here now?

File: src/harness/unittests/session.ts [hunk 2]
```diff
@@ -16,8 +16,8 @@ namespace ts.server {
         directoryExists: () => false,
         getDirectories: () => [],
         createDirectory: noop,
-        getExecutingFilePath(): string { return void 0; },
-        getCurrentDirectory(): string { return void 0; },
+        getExecutingFilePath(): string { return ""; },
+        getCurrentDirectory(): string { return ""; },
```
Comment (billti) [inline src/harness/unittests/session.ts]: Deliberate?

Comment (RyanCavanaugh) [reply src/harness/unittests/session.ts]: Yes

File: tests/baselines/reference/api/tsserverlibrary.d.ts (hunk 1 above)
Comment (RyanCavanaugh) [reply tests/baselines/reference/api/tsserverlibrary.d.ts]: Got inadvertently picked up during a branch switch; will remove

File: src/harness/unittests/session.ts (hunk 2 above)
Comment (RyanCavanaugh) [reply src/harness/unittests/session.ts]: Clarification: This is what it does in `master`

Comment (billti) [reply src/harness/unittests/session.ts]: Right... but is it in any way related to the fixes happening here? If not, is there any reason to need this change also here?

Comment (billti) [reply src/harness/unittests/session.ts]: Actually.... it's in the harness, not the product, so if the tests are passing I don't really care :-p

Comment (RyanCavanaugh) [reply src/harness/unittests/session.ts]: We need this because we compute the default safe list location based on the current directory / current executing file path
